pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1426,6 +1470,21 @@ public LLRealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableCo
     try {
       _partitionGroupConsumerSemaphore.acquire();
       _acquiredConsumerSemaphore.set(true);
+      if (_tableConfig.getUpsertConfig() != null && _tableConfig.getUpsertConfig().isEnableSnapshot()) {
+        // block ingestion for new consuming segments
+        _isReadyToConsumeData = () -> false;
```

Comment:
so this will always override the `isReadyToConsumeData ` to `LLRealtimeSegmentDataManager `. Is there any case that we want to use input param?